src/backend/handler/gamestatus/gamestatushandler.ts

```diff
@@ -0,0 +1,94 @@
+import { GameStatus } from './../../../common/types'
+import { getMainWindow } from '../../utils'
+import Store from 'electron-store'
+
+const store = new Store({
+  cwd: 'store',
+  name: 'progress-handler'
+})
+
+function initGameStatusHandler() {
+  const handlers = getAllGameStatus()
+  handlers.forEach((status) => {
+    status.previousProgress = status.progress
+    status.status = 'done'
+  })
+
+  store.set('gameStatus', handlers)
+}
+
+function setGameStatusOfElement(status: GameStatus) {
+  let newStatus = status
+  if (!store.has('gameStatus')) {
+    store.set('gameStatus', [status])
+    return
+  }
+
+  const handlers = store.get('gameStatus') as GameStatus[]
```

Comment:
I wonder if we can actually store an object with `{appName => GameStatus}` pairs instead, it should simplify a lot of the adding/finding/removing (you would be able to remove all the filtering, indexes, slice, etc).

We can always go back to a list of game statuses doing `Object.values(handlers)` if we need to for example.

Reply:
I tried with Map, but map can't be stored without convertion into electron-store.
I don't understand the object pairs. Can you give me an example ?

Reply:
I mean to store something like:

```js
{
  201302193029103920301923: {
    appName: 201302193029103920301923,
    progress: ...,
    status: ...,
    etc: ...
  },
  anotherAppName: {
    appName: anotherAppName,
    theRestOfTheFields: ...
  }
}
```

That can't be stored directly?